test/commons/dom/get-parsed-resource.js

```diff
@@ -0,0 +1,105 @@
+describe('dom.getParsedResource', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var queryFixture = axe.testUtils.queryFixture;
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+	});
+
+	it('returns undefined when given node does not have specified attribute', function() {
+		var vNode = queryFixture(
+			'<button id="target" role="link">Schedule appointment</button>'
+		);
+		var actual = axe.commons.dom.getParsedResource(vNode.actualNode, 'href');
+		assert.isUndefined(actual);
+	});
+
+	it('returns undefined when `A` has no `HREF` attribute', function() {
+		var vNode = queryFixture('<a id="target">Follow us on Instagram</a>');
+		var actual = axe.commons.dom.getParsedResource(vNode.actualNode, 'href');
+		assert.isUndefined(actual);
+	});
+
+	it('returns parsed resource when `A` has empty `HREF`', function() {
+		var vNode = queryFixture(
+			'<a id="target" href="">Follow us on Instagram</a>'
+		);
+		var actual = axe.commons.dom.getParsedResource(vNode.actualNode, 'href');
+		assert.isDefined(actual);
+		assert.isUndefined(actual.hash);
+		assert.isUndefined(actual.filename);
+		assert.equal(actual.hostname, 'localhost');
+		assert.equal(actual.pathname, '/test/commons');
+	});
+
+	it('returns parsed resource for `A` with `HREF`', function() {
+		var vNode = queryFixture(
+			'<a id="target" href="https://facebook.com">follow us on Facebook</a>'
+		);
+		var actual = axe.commons.dom.getParsedResource(vNode.actualNode, 'href');
+		assert.isDefined(actual);
+		assert.equal(actual.protocol, 'https:');
+		assert.equal(actual.hostname, 'facebook.com');
```

Comment:
I think you should do a `deepEqual` on actual, so that you don't get unexpected props from your test.